File: OCR-Translate/Models/AnalysisResult.swift
import Foundation

enum AnalysisMode: String, CaseIterable, Equatable {
    case detailed = "详细分析"
    case concise = "简洁分析"
}

struct VocabularyAnnotation: Identifiable, Equatable {
    let id = UUID()
    let word: String
    let reading: String
    let meaning: String
    let partOfSpeech: String?
    let jlptLevel: String?
    let notes: String?
}

struct SentenceComponent: Identifiable, Equatable {
    let id = UUID()
    let label: String
    let text: String
    let explanation: String?
}

struct SentenceAnalysis: Identifiable, Equatable {
    let id = UUID()
    let originalSentence: String
    let components: [SentenceComponent]
    let grammarPoints: [String]
    let vocabulary: [VocabularyAnnotation]
}

struct AnalysisResult: Equatable {
    let mode: AnalysisMode
    let sentences: [SentenceAnalysis]
    let overallNotes: String?
}

File: win/NGMproofread.Windows/Models/AnalysisResult.cs
namespace NGMproofread.Windows.Models;

public enum AnalysisMode
{
    Proofread,
    Study
}

public static class AnalysisModeExtensions
{
    public static string DisplayName(this AnalysisMode mode) => mode switch
    {
        AnalysisMode.Proofread => "校对模式",
        AnalysisMode.Study => "学习模式",
        _ => ""
    };
}

public class VocabularyAnnotation
{
    public string Id { get; init; } = Guid.NewGuid().ToString();
    public string Word { get; init; } = "";
    public string Reading { get; init; } = "";
    public string Meaning { get; init; } = "";
    public string? PartOfSpeech { get; init; }
    public string? JlptLevel { get; init; }
    public string? Notes { get; init; }
}

public class SentenceComponent
{
    public string Id { get; init; } = Guid.NewGuid().ToString();
    public string Label { get; init; } = "";
    public string Text { get; init; } = "";
    public string? Explanation { get; init; }
    public List<SentenceComponent> Children { get; init; } = [];
}

public class SentenceAnalysis
{
    public string Id { get; init; } = Guid.NewGuid().ToString();
    public string OriginalSentence { get; init; } = "";
    public List<SentenceComponent> Components { get; init; } = [];
    public List<string> GrammarPoints { get; init; } = [];
    public List<VocabularyAnnotation> Vocabulary { get; init; } = [];
}

public class AnalysisResult
{
    public AnalysisMode Mode { get; init; }
    public List<SentenceAnalysis> Sentences { get; init; } = [];
    public string? OverallNotes { get; init; }
}
